drivers/azure/azure.go

```diff
@@ -373,6 +389,21 @@ func (d *Driver) PreCreateCheck() (err error) {
 		}
 	}
 
+	if d.AvailabilityZone != "" {
+		if !d.ManagedDisks {
+			return fmt.Errorf("Managed Disks must be used when creating resources in specific Availability Zones (--azure-managed-disks)")
+		}
+		if v, err := strconv.Atoi(d.AvailabilityZone); err != nil || v == 0 {
+			return fmt.Errorf("Each VM can only be assigned to a single Availability Zone. Each zone is denoted by an integer greater than 0")
+		}
+		if !d.EnablePublicIPStandardSKU {
+			return fmt.Errorf("The Standard Public IP SKU must be enabled when creating resources in specific Availablity Zones (--azure-enable-public-ip-standard-sku)")
+		}
+		if d.AvailabilitySet != "" {
+			log.Warn("Both an Availability Set and an Availability Zone were specified. Skipping creation of the Availability Set, only creating resources in the specified Availability Zone.")
```

Comment:
Nit: long log message. You could shorten to `Both an Availability Set and Availability Zone were specified. Skipping creating the Availability Set, only creating resources in the Availability Zone.`

Reply:
Personally I feel that it's more important to have a sentence that can be read fluidly over one that is slightly shorter. I will shorten the first sentence, but I'm not sure there is a hard rule about this and would prefer to keep the second sentence as-is.